src/content/1.7/development/components/grid/_index.md

```diff
@@ -1,324 +1,410 @@
+    public: true
+``` 
 
-## Grid Data Factory
+Most of the time you won't be creating Grid Definition by yourself but delegating this task to other services.
+But in case you need to create Grid Definition by hand, here's how you can do that.
 
-As you can imagine, the responsibility of the Grid Data Factory is to retrieve the Grid data, based on the Grid Definition and the Grid Query Builder.
+```php
+$productsGridDefinitionFactory = $container->get('prestashop.core.grid.definition.factory.product_grid_definition_factory');
+$productsGridDefinition = $productsGridDefinitionFactory->getDefinition();
 
-The only method available of `GridDataFactoryInterface` is `getData` which returns an instance of `GridData`.
-A GridData is an **immutable object** used to store and retrieve the grid data, so if you want to alter this data, you must do it in Grid Data Factory *before* the GridData creation.
+// you can access all information thats was defined
+$productsGridDefinition->getColumns(); // collection of defined columns
+$productsGridDefinition->getName(); // "Products"
+$productsGridDefinition->getId(); // "products"
+``` 
 
-You can rely on the implementation provided in the component `GridDataFactory` or create your own if you have specific needs. 
+## Search Criteria
 
-## Grid Factory
+In Grid component Search Criteria is used for Grid's data sorting, paginating & filtering. Search Criteria
+can be loaded from database, URL query or anywhere else. 
 
-Finally the Grid Factory is the final top layer above the other ones, it retrieves the Grid Data thanks to the Grid Data Factory and builds a form to manage filters based on the information in the Grid Definition. Its interface `GridFactoryInterface` implements one method `getGrid`.
+{{% notice info %}}
+Grid component itself does not manage Search Criteria but instead it provides interface for it.
+In PrestaShop [Filters component](#) is used to resolve Search Criteria for Grid.
+{{% /notice %}}
 
-Just like the Grid Data Factory you can either implement your own interface or use the provided `GridFactory` provided in Prestashop core code.
+{{% notice note %}}
+Search Criteria is immutable. This means that once Search Criteria is created it cannot be changed.
+{{% /notice %}}
 
-## Grid services declaration
+### Creating Search Criteria
 
-Only 4 services must be declared: the *Grid Definition Factory*, the *Grid Query Builder*, the *Grid Data Factory*, and the *Grid Factory*.
+Even though most of the time Search Criteria will be created using [Filters component](#), you can still
+create it by yourself. Grid provides simple implementation for it.
 
-```yaml
-# In src/PrestaShopBundle/Resources/config/services/core/grid.yml
-parameters:
-  prestashop.core.grid.data.factory.doctrine_grid_data_factory: PrestaShop\PrestaShop\Core\Grid\Data\Factory\DoctrineGridDataFactory
+```php
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteria;
+
```

Comment:
this class does not exist in 1.7.5, should we specify the minimum version to use this class?